### PR TITLE
[New Hunt] Add Initial Linux Hunting Files

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/ia-trade-team/issues/280

## Summary
This pull request adds initial Linux hunting files created by @Aegrah, both TOML and MD.

### CODE REVIEW

**Comment by Aegrah** [comment]
## Update
in commit https://github.com/elastic/detection-rules/pull/3847/commits/9a659596aed33dec8f29e37d00b8a788fc86eb6f I:
- Added list of strings to add ESQL next to SQL (for OSQuery).
- Removed 2 hunts related to web server activity that require more research before being published.
- Changed a few file names based on MITRE.
- Altered a few hunt logic sections, descriptions, notes and more.
- Reran the markdown generation and committed. It ended up changing Windows rules as well, which I don't know why exactly. CC @terrancedejesus, if something went wrong here, we need to fix or revert this commit.

Just for reference, the only command that was run:
```
ruben@pc01:/mnt/c/Users/ruben/Documents/GitHub/detection-rules/hunting$ python3 generate_markdown.py
Markdown generated: /mnt/c/Users/ruben/Documents/GitHub/detection-rules/hunting/linux/docs/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.md
Markdown generated: /mnt/c/Users/ruben/Documents/GitHub/detection-rules/hunting/linux/docs/command_and_control_via_unusual_file_downloads_from_source_addresses.md
[...]
```

Thanks for getting this amazing PR in so quickly Terrance!

**Comment by terrancedejesus** [comment]
> Reran the markdown generation and committed. It ended up changing Windows rules as well, which I don't know why exactly. CC @terrancedejesus, if something went wrong here, we need to fix or revert this commit.

@Aegrah This happens because the generate markdown file re-generates all MD files at the moment. It is on the roadmap to be refined somewhat for more flexibility in the near future.

**Comment by terrancedejesus** [comment]
@Aegrah - I went ahead and updated all the Windows' rules to have the right schema and then re-generated all the docs. I updated the index as well and made some minor changes for generating the markdown.

**Review by Aegrah — APPROVED** [review]
Looks good to me!

File: hunting/linux/docs/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+- **Integration:** [endpoint](https://docs.elastic.co/integrations/endpoint)
+- **Language:** `[ES|QL]`
+
+## Query
+
+```sql
+from logs-endpoint.events.network-*
+| where @timestamp > now() - 7 day
+| where host.os.type == "linux" and event.type == "start" and event.action in ("connection_attempted", "connection_accepted") and destination.ip IS NOT null and not CIDR_MATCH(destination.ip, "10.0.0.0/8", "127.0.0.0/8", "169.254.0.0/16", "172.16.0.0/12", "192.0.0.0/24", "192.0.0.0/29", "192.0.0.8/32", "192.0.0.9/32", "192.0.0.10/32", "192.0.0.170/32", "192.0.0.171/32", "192.0.2.0/24", "192.31.196.0/24", "192.52.193.0/24", "192.168.0.0/16", "192.88.99.0/24", "224.0.0.0/4", "100.64.0.0/10", "192.175.48.0/24","198.18.0.0/15", "198.51.100.0/24", "203.0.113.0/24", "224.0.0.0/4", "240.0.0.0/4", "::1","FE80::/10", "FF00::/8")
+| stats process_count = count(process.name), agent_count = count_distinct(agent.id) by process.name
```
Comment (Samirbous) [inline hunting/linux/docs/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.md:20]: this doesn't make sense count(process.name) ... and by process.name (the count will be alway equal to 1)

File: hunting/linux/queries/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.toml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+[hunt]
+author = "Elastic"
+description = """
+This hunt identifies network connections with low occurrence frequency for unique agent IDs on Linux systems. It monitors network connection attempts and acceptances, focusing on those initiated by processes that are rarely seen in the environment. By accounting for known low-frequency legitimate binaries (LoLBins) and suspicious directories, this hunt aims to detect unusual network activity that may indicate malicious behavior.
+"""
+integration = ["endpoint"]
+uuid = "q1a1d7n7-7890-4q1q-a91q-be456ab80o89"
```
Comment (w0rk3r) [inline hunting/linux/queries/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.toml:7]: .

File: hunting/linux/queries/persistence_via_unusual_system_binary_parent.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "Requires additional data analysis and investigation into results to identify malicious or unauthorized use of system binaries."
+]
+mitre = ["T1546.004", "T1059.004"]
+
+query = [
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 30 day
+| where host.os.type == "linux" and event.action == "exec" and event.type == "start" and process.parent.name in ("ls", "cat", "mkdir", "touch", "mv", "cp")
+| stats process_cli_count = count(process.command_line), process_count = count(process.executable), host_count = count_distinct(host.name) by process.parent.executable, process.executable
```
Comment (Samirbous) [inline hunting/linux/queries/persistence_via_unusual_system_binary_parent.toml:24]: same issue `count(field) by field` will always return 1.

File: hunting/linux/queries/privilege_escalation_via_process_capabilities.toml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 30 day
+| where host.os.type == "linux" and event.action == "exec" and event.type == "start" and (process.thread.capabilities.effective is not null or process.thread.capabilities.permitted is not null) and user.id != "0" and
+not (
+  // Remove these if you expect persistence through capabilities
+  process.executable like "/var/lib/docker/*" or
+  process.name == "gnome-keyring-daemon" or
+  process.thread.capabilities.permitted == "CAP_WAKE_ALARM"
+)
+| stats process_cli_count = count(process.command_line), process_count = count(process.executable), host_count = count_distinct(host.name) by process.parent.executable, process.executable, process.command_line, process.thread.capabilities.effective, process.thread.capabilities.permitted, user.id
```
Comment (Samirbous) [inline hunting/linux/queries/privilege_escalation_via_process_capabilities.toml:31]: same issue `count(field) by field` will always return 1.

File: hunting/linux/queries/privilege_escalation_via_process_capabilities.toml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+| limit 100
+''',
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 90 day
+| where host.os.type == "linux" and event.action == "exec" and event.type == "start" and (
+  process.thread.capabilities.effective in ("CAP_SYS_MODULE", "CAP_SYS_PTRACE", "CAP_DAC_OVERRIDE", "CAP_DAC_READ_SEARCH", "CAP_SETUID", "CAP_SETGID", "CAP_SYS_ADMIN") or
+  process.thread.capabilities.permitted in ("CAP_SYS_MODULE", "CAP_SYS_PTRACE", "CAP_DAC_OVERRIDE", "CAP_DAC_READ_SEARCH", "CAP_SETUID", "CAP_SETGID", "CAP_SYS_ADMIN")
+) and user.id != "0"
+| stats process_cli_count = count(process.command_line), process_count = count(process.executable), host_count = count_distinct(host.name) by process.parent.executable, process.executable, process.command_line, process.thread.capabilities.effective, process.thread.capabilities.permitted, user.id
```
Comment (Samirbous) [inline hunting/linux/queries/privilege_escalation_via_process_capabilities.toml:43]: same issue `count(field) by field` will always return 1.

File: hunting/linux/queries/privilege_escalation_via_segmentation_fault_and_buffer_overflow.toml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+| keep timestamp, process, pid, so_file, segfault_address, instruction_pointer, stack_pointer, error_code, so_base_address, so_offset
+''',
+'''
+from logs-system.syslog*
+| where host.os.type == "linux" and process.name == "kernel" and message like "*segfault*"
+| where @timestamp > now() - 12 hour
+| grok message "\\[%{DATA:timestamp}\\] %{WORD:process}\\[%{NUMBER:pid}\\]: segfault at %{BASE16NUM:segfault_address} ip %{BASE16NUM:instruction_pointer} sp %{BASE16NUM:stack_pointer} error %{NUMBER:error_code} in %{DATA:so_name}\\[%{BASE16NUM:so_base_address}\\+%{BASE16NUM:so_offset}\\] likely on CPU %{NUMBER:cpu} \\(core %{NUMBER:core}, socket %{NUMBER:socket}\\)"
+| eval timestamp = REPLACE(timestamp, "\\s+", "")
+| keep timestamp, process, pid, segfault_address, instruction_pointer, stack_pointer, error_code, so_name, so_base_address, so_offset, cpu, core, socket
+| stats process_count = count(process), so_count = count(so_name) by process, so_name
```
Comment (Samirbous) [inline hunting/linux/queries/privilege_escalation_via_segmentation_fault_and_buffer_overflow.toml:35]: same issue `count(field) by field` will always return 1.

**Comment by terrancedejesus** [comment]
After further discussion @Aegrah will update some of queries and their aggregations on July 3, 2024 based on a conversation with @Samirbous about the `COUNT()` functionality in ES|QL.

**Comment by terrancedejesus** [comment]
Queries have been adjusted by @Aegrah and docs regenerated. We also fixed the UUIDs to only include hex characters.
@Samirbous @w0rk3r can you take another look when time permits?

File: hunting/linux/queries/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.toml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+| stats cc = count(), agent_count = count_distinct(agent.id) by process.name
+| where agent_count <= 3 and cc > 0 and cc <= 5
+| limit 100
+| sort cc asc
+''',
+'''
+from logs-endpoint.events.network-*
+| where @timestamp > now() - 30 day
+| where host.os.type == "linux" and event.type == "start" and event.action in ("connection_attempted", "connection_accepted") and (
+    (process.executable like "./") or
```
Comment (Samirbous) [inline hunting/linux/queries/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.toml:50]: missing wildcard ? 

```suggestion
    (process.executable like "./*") or
```

File: hunting/linux/queries/command_and_control_via_unusual_file_downloads_from_source_addresses.toml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    "The process command line count threshold is set to <= 10, and the host count threshold is set to <= 5 to balance detection and noise."
+]
+mitre = ["T1071.001", "T1071.004"]
+
+query = [
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 7 day
+| where host.os.type == "linux" and event.type == "start" and process.name in ("curl", "wget") and process.command_line rlike """.*[0-9]{1,3}(\.[0-9]{1,3}){3}.*"""
+| stats cc = count(), host_count = count(host.name) by process.command_line, process.executable, host.name
```
Comment (Samirbous) [inline hunting/linux/queries/command_and_control_via_unusual_file_downloads_from_source_addresses.toml:24]: count(host.name) by host.name ? 

```suggestion
| stats cc = count(), host_count = count_distinct(host.id) by process.command_line, process.executable
```

File: hunting/linux/queries/defense_evasion_via_capitalized_process_execution.toml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+query = [
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 10 day
+| where host.os.type == "linux" and event.type == "start" and event.action == "exec" and (
+  (process.name rlike """[A-Z]{2,}[a-z]{1,}[0-9]{0,}""") or
+  (process.name rlike """[A-Z]{1,}[0-9]{0,}""")
+)
+| stats cc = count(), host_count = count(host.name) by process.name
```
Comment (Samirbous) [inline hunting/linux/queries/defense_evasion_via_capitalized_process_execution.toml:26]:
```suggestion
| stats cc = count(), host_count = count_distinct(host.name) by process.name
```

File: hunting/linux/queries/defense_evasion_via_hidden_process_execution.toml [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+]
+mitre = ["T1036.004", "T1059"]
+
+query = [
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 180 day
+| where host.os.type == "linux" and event.type == "start" and event.action == "exec" and (
+  (process.executable rlike "/[^/]+/\\.[^/]+")
+)
```
Comment (Samirbous) [inline hunting/linux/queries/defense_evasion_via_hidden_process_execution.toml:23]: unnecessary parenthesis ? 

```suggestion
| where host.os.type == "linux" and event.type == "start" and event.action == "exec" and 
  process.executable rlike "/[^/]+/\\.[^/]+"
```

File: hunting/linux/queries/defense_evasion_via_hidden_process_execution.toml [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+| stats cc = count(), host_count = count(host.name) by process.executable, process.parent.executable, host.name, user.id
```
Comment (Samirbous) [inline hunting/linux/queries/defense_evasion_via_hidden_process_execution.toml:24]: the cc = count() should be equal to host_count = count(host.name) by host.name ? 

```suggestion
| stats cc = count(), host_count = count_distinct(host.name) by process.executable, process.parent.executable, user.id
```

File: hunting/linux/queries/defense_evasion_via_multi_dot_process_execution.toml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    "The process count threshold of <= 10 can be adjusted based on the environment's baseline activity."
+]
+mitre = ["T1036.004", "T1070"]
+
+query = [
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 30 day
+| where host.os.type == "linux" and event.type == "start" and event.action == "exec" and process.executable rlike """.*\.{3,}.*"""
+| stats cc = count(), host_count = count(host.name) by process.executable
```
Comment (Samirbous) [inline hunting/linux/queries/defense_evasion_via_multi_dot_process_execution.toml:22]: why not count_distinct to get idea of distribution per host vs duplicate events (e.g 10 times from same host vs 10 times on 10 unique hosts)

```suggestion
| stats cc = count(), host_count = count_distinct(host.name) by process.executable
```

File: hunting/linux/queries/defense_evasion_via_multi_dot_process_execution.toml [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+// Alter this threshold to make sense for your environment
+| where cc <= 10
```
Comment (Samirbous) [inline hunting/linux/queries/defense_evasion_via_multi_dot_process_execution.toml:24]: host_count is not used in the filtering logic ?

File: hunting/linux/queries/excessive_ssh_network_activity_unique_destinations.toml [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+]
+mitre = ["T1021.004", "T1078.003"]
+
+query = [
+'''
+from logs-endpoint.events.network-*
+| where @timestamp > now() - 7 day
+| where host.os.type == "linux" and event.category == "network" and network.transport == "tcp" and destination.port == 22 and source.port >= 49152
+| keep destination.ip, host.id, user.name
+| stats count_unique_dst = count_distinct(destination.ip) by host.id, user.name
```
Comment (Samirbous) [inline hunting/linux/queries/excessive_ssh_network_activity_unique_destinations.toml:24]: host.id is necessary here or just user.name ?

File: hunting/linux/queries/execution_uncommon_process_execution_from_suspicious_directory.toml [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+  (process.executable rlike "/run/[^/]+") or
+  (process.executable rlike "/var/run/[^/]+")
+) and not (
+  // Exclude noisy (parent) processes, users or directories from your environment here
+  (process.parent.executable in ("/usr/sbin/dpkg-preconfigure")) or
+  // Exclude /tmp and /var/tmp instances starting or ending with digits (usually benign files)
+  (process.executable rlike "/tmp/[0-9].*" or process.executable rlike "/tmp/.*[0-9]/?") or
+  (process.executable rlike "/var/tmp/[0-9].*" or process.executable rlike "/var/tmp/.*[0-9]/?")
+)
+| stats cc = count(), host_count = count(host.name) by process.executable, process.parent.executable, host.name, user.id
```
Comment (Samirbous) [inline hunting/linux/queries/execution_uncommon_process_execution_from_suspicious_directory.toml:38]: the more you reduce by a, b, c, d to the minimum the more you improve performance. also aggregating by unique fields like user.name, source.ip etc is good if it align with logic like top noisy users or top targeted destinations, in this rule it's about the process.executable. 

```suggestion
| stats cc = count(), host_count = count_distinct(host.name) by process.executable, process.parent.executable
```

File: hunting/linux/queries/low_volume_modifications_to_critical_system_binaries.toml [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+  (file.path like "/usr/bin/*") or
+  (file.path like "/sbin/*") or
+  (file.path like "/usr/sbin/*")
+) and not (
+  // Exclude expected update processes, e.g., package managers
+  (process.executable in ("/usr/bin/apt", "/usr/bin/dpkg", "/usr/bin/yum", "/usr/bin/rpm", "/usr/bin/pacman", "/usr/bin/pamac-daemon", "/usr/bin/update-alternatives", "/usr/bin/dockerd", "/usr/bin/microdnf", "/sbin/apk")) or
+  // Exclude certain benign or expected modification patterns, if applicable
+  (file.path like "/usr/bin/gzip*") // Example exclusion, adjust based on your environment
+)
+| stats modification_count = count(file.path), unique_files_modified = count_distinct(file.path), host_count = count(host.name) by process.executable, host.name, user.name
```
Comment (Samirbous) [inline hunting/linux/queries/low_volume_modifications_to_critical_system_binaries.toml:34]: count(host.name) by host.name

File: hunting/linux/queries/persistence_via_cron.toml [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+| stats pers_count = count(persistence), agent_count = count(agent.id) by process.executable, file.path, host.name, user.name
+| where pers_count > 0 and pers_count <= 20 and agent_count <= 3
+| sort pers_count asc
+| limit 100
+''',
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 30 day
+| where host.os.type == "linux" and event.action == "exec" and event.type == "start" and process.parent.name in ("cron", "fcron", "atd")
+| stats cc = count(), host_count = count_distinct(host.name) by process.command_line, process.executable, process.parent.executable
```
Comment (Samirbous) [inline hunting/linux/queries/persistence_via_cron.toml:58]: optional but host.id looks more unique than host.name

File: hunting/linux/queries/persistence_via_driver_load_with_low_occurrence_frequency.toml [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+    "Such rare driver loads can indicate potentially malicious activity or the presence of uncommon drivers."
+]
+mitre = ["T1547.006", "T1069.002"]
+
+query = [
+'''
+from logs-auditd_manager.auditd-*, logs-auditd.log-*, auditbeat-*
+| where @timestamp > now() - 365 day
+| where host.os.type == "linux" and event.category == "driver" and event.action == "loaded-kernel-module" and auditd.data.syscall in ("init_module", "finit_module")
+| stats host_count = count_distinct(host.id), total_count = count(*), ko_count = count_distinct(auditd.data.name) by auditd.data.name, process.executable, process.name
```
Comment (Samirbous) [inline hunting/linux/queries/persistence_via_driver_load_with_low_occurrence_frequency.toml:23]: count_distinct(auditd.data.name) by auditd.data.name ?

File: hunting/linux/queries/persistence_via_message_of_the_day.toml [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+| where pers_count > 0 and pers_count <= 20 and agent_count <= 5
+| sort pers_count asc
+| limit 100
+''',
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 30 day
+| where host.os.type == "linux" and event.action == "exec" and event.type == "start" and process.parent.executable like "/etc/update-motd.d/*" and
+not process.args like "/tmp/tmp.*"
+| stats cc = count(), host_count = count_distinct(host.name) by process.command_line, process.executable, process.parent.executable
```
Comment (Samirbous) [inline hunting/linux/queries/persistence_via_message_of_the_day.toml:40]: by cmdline could be bad for perf

File: hunting/linux/queries/privilege_escalation_via_segmentation_fault_and_buffer_overflow.toml [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+]
+mitre = ["T1203", "T1068"]
+
+query = [
+'''
+from logs-system.syslog*
+| where @timestamp > now() - 12 hour
+| where host.os.type == "linux" and process.name == "kernel" and message like "*segfault*"
+| grok message "\\[%{NUMBER:timestamp}\\] %{WORD:process}\\[%{NUMBER:pid}\\]: segfault at %{BASE16NUM:segfault_address} ip %{BASE16NUM:instruction_pointer} sp %{BASE16NUM:stack_pointer} error %{NUMBER:error_code} in %{DATA:so_file}\\[%{BASE16NUM:so_base_address}\\+%{BASE16NUM:so_offset}\\]"
+| keep timestamp, process, pid, so_file, segfault_address, instruction_pointer, stack_pointer, error_code, so_base_address, so_offset
```
Comment (Samirbous) [inline hunting/linux/queries/privilege_escalation_via_segmentation_fault_and_buffer_overflow.toml:26]: no filtering or stats needed ?

File: hunting/linux/queries/privilege_escalation_via_process_capabilities.toml [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 30 day
+| where host.os.type == "linux" and event.action == "exec" and event.type == "start" and (process.thread.capabilities.effective is not null or process.thread.capabilities.permitted is not null) and user.id != "0" and
+not (
+  // Remove these if you expect persistence through capabilities
+  process.executable like "/var/lib/docker/*" or
+  process.name == "gnome-keyring-daemon" or
+  process.thread.capabilities.permitted == "CAP_WAKE_ALARM"
+)
+| stats cc = count(), host_count = count_distinct(host.name) by process.parent.executable, process.executable, process.command_line, process.thread.capabilities.effective, process.thread.capabilities.permitted, user.id
```
Comment (Samirbous) [inline hunting/linux/queries/privilege_escalation_via_process_capabilities.toml:31]: can't we limit this to by process.executbale and capabilities (too many aggregation pts reduce perf)

File: hunting/linux/queries/privilege_escalation_via_segmentation_fault_and_buffer_overflow.toml (hunk 20 above)
Comment (Aegrah) [reply hunting/linux/queries/privilege_escalation_via_segmentation_fault_and_buffer_overflow.toml:26]: Nope there's two queries, one hunting for segfaults and the other one for stats, depending on the use case.

**Comment by Aegrah** [comment]
## Review
I did another full review:
- Removed unnecessary group by's and evaluated the results;
- Added `count_distinct` to the host counting where necessary;
- Removed unnecessary parentheses;
- Updated some query formatting;
- Checked all of the updated queries to see whether they still worked.

File: hunting/llm/docs/aws_bedrock_dos_resource_exhaustion_detection.md [hunk 22]
```diff
@@ -7,9 +7,9 @@
 - **Author:** Elastic
 - **Description:** This hunting query identifies unusual spikes in token usage that may indicate malicious attempts to disrupt services. High token usage can strain system resources and degrade performance, aligning with tactics observed in DoS attacks.
 
-- **UUID:** `dc181967-c32c-46c9-b84b-ec4c8811c6a0`
+- **UUID:** `00023411-192e-4472-90aa-da7562bc3f2a`
```
Comment (Mikaayenson) [inline hunting/llm/docs/aws_bedrock_dos_resource_exhaustion_detection.md:10]: @terrancedejesus why were these modified?

File: hunting/linux/queries/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.toml [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+    "Identifies suspicious directories where processes are executed from, which can indicate malicious activity."
+]
+mitre = ["T1071.001", "T1071.004"]
+
+query = [
+'''
+from logs-endpoint.events.network-*
+| where @timestamp > now() - 7 day
+| where host.os.type == "linux" and event.type == "start" and event.action in ("connection_attempted", "connection_accepted") and destination.ip IS NOT null and not CIDR_MATCH(destination.ip, "10.0.0.0/8", "127.0.0.0/8", "169.254.0.0/16", "172.16.0.0/12", "192.0.0.0/24", "192.0.0.0/29", "192.0.0.8/32", "192.0.0.9/32", "192.0.0.10/32", "192.0.0.170/32", "192.0.0.171/32", "192.0.2.0/24", "192.31.196.0/24", "192.52.193.0/24", "192.168.0.0/16", "192.88.99.0/24", "224.0.0.0/4", "100.64.0.0/10", "192.175.48.0/24","198.18.0.0/15", "198.51.100.0/24", "203.0.113.0/24", "224.0.0.0/4", "240.0.0.0/4", "::1","FE80::/10", "FF00::/8")
+| stats cc = count(), agent_count = count_distinct(agent.id) by process.name
```
Comment (Samirbous) [inline hunting/linux/queries/command_and_control_via_network_connections_with_low_occurrence_frequency_for_unique_agents.toml:24]: process path gives more clues to pivot vs process.name 

```suggestion
| stats cc = count(), agent_count = count_distinct(agent.id) by process.executable
```

**Review by Samirbous — APPROVED** [review]
left few more comments, tested them against 8.14 no synthax errors, will be good to test them to see if they return the expected data. other than that I'm fine with merging, good set of hunts!

File: hunting/llm/docs/aws_bedrock_dos_resource_exhaustion_detection.md (hunk 22 above)
Comment (Mikaayenson) [reply hunting/llm/docs/aws_bedrock_dos_resource_exhaustion_detection.md:10]: inadvertently updated with automation. should be g2g assuming we dont mass change uuids in the future.

**Review by Mikaayenson — APPROVED** [review]
ideally @Aegrah, consider the last round of feedback before pushing it. Otherwise, lgtm.

**Comment by Aegrah** [comment]
Thanks for all of the thorough reviews @Samirbous. Just pushed an update which added one rule (hunting for bind/reverse shells via OSQuery), and incorporated the feedback from Samir. Once the docs are regenerated it is good to go IMO @terrancedejesus. CC @Mikaayenson Thanks all!

**Review by eric-forte-elastic — APPROVED** [review]

**Pull request merged**